FHEM/10_BindingsIo.pm

```diff
@@ -85,40 +97,42 @@ BindingsIo_Define($$$)
 
   if ($init_done && $localServer == 1) {
     my $foundServer = 0;
-    foreach my $fhem_dev (sort keys %main::defs) {
-      $foundServer = 1 if($main::defs{$fhem_dev}{TYPE} eq $bindingType."Binding");
-      $foundServer = 1 if($main::defs{$fhem_dev}{TYPE} eq $bindingType."Server");
-    }
-    if ($foundServer == 0) {
-      CommandDefine(undef, $bindingType."server_".$hash->{PORT}." ".$bindingType."Server ".$port);
-      InternalTimer(gettimeofday()+3, "BindingsIo_connectDev", $hash, 0);
+    
+    # use devspec2array to find existing fhempy server device
+    my @fhempyDevices;
+    @fhempyDevices = ::devspec2array(qq[i:TYPE=$bindingType(Binding|Server)]);
+    $foundServer = ::IsDevice($fhempyDevices[0]);
+
+    if ($foundServer == 0) 
```

Comment:
Es kann sein, dass BindingsIo vor dem Server in der fhem.cfg hinterlegt ist, daher der Check nur bei `$init_done`, ansonsten wird der Server bei jedem Neustartet versucht anzulegen.

Reply:
Ok das verstehe ich mit dem `$init_done`.

Was ich allerdings nicht verstehe, woher kommt der Type `fhempyBinding`. Gibt es den, wenn ja wo ist er definiert?
`fhempyServer` gibt es.

Reply:
`fhempyBinding` gibt/gab es nie. Nur `PythonBinding` gab es. Der Check mit $bindingtype liegt nur daran, dass es auch für nicht Python (z.B. nodejs) funktionieren soll.

Reply:
auf `init_done` wird schon in Zeile 100 geprüft :)

Braucht es hier jetzt noch eine Anpassung?

Reply:
Stimmt, das hatte ich übersehen!